library/std/src/sys/unix/fs.rs

```diff
@@ -1126,6 +1126,19 @@ impl fmt::Debug for File {
             Some(PathBuf::from(OsString::from_vec(buf)))
         }
 
+        #[cfg(target_os = "freebsd")]
+        fn get_path(fd: c_int) -> Option<PathBuf> {
+            let info = Box::<libc::kinfo_file>::new_zeroed();
+            let mut info = unsafe { info.assume_init() };
+            (*info).kf_structsize = mem::size_of::<libc::kinfo_file>() as libc::c_int;
+            let n = unsafe { libc::fcntl(fd, libc::F_KINFO, *info) };
```

Comment:
I think the argument to fcntl here should be info.as_mut_ptr(), not `*info`, which presumably is passing info by-value rather than a pointer to it.

Reply:
Are you sure we can call as_mut_ptr at this point (after assume_init) ?

Reply:
Oh, sorry, `&mut *info as *mut libc::kinfo_file`. Looks like Box itself doesn't have a way to get at the pointer via a function.